smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/impl/ConfiguredChannelFactory.java

```diff
@@ -142,7 +144,13 @@ private static String getConnectorAttribute(Config config) {
         IncomingConnectorFactory mySourceFactory = incomingConnectorFactories.select(ConnectorLiteral.of(connector))
                 .stream().findFirst().orElseThrow(() -> new IllegalArgumentException("Unknown connector for " + name + "."));
 
-        return mySourceFactory.getPublisherBuilder(config);
+        PublisherBuilder<? extends Message> publisher = mySourceFactory.getPublisherBuilder(config);
+
+        for (PublisherDecorator decorator : CDI.current().select(PublisherDecorator.class)) {
```

Comment:
For my own information (remember, I'm a CDI noob), what's the difference between this and injecting an `Instance` and iterating on the set of providers?

Reply:
There's no difference, except that `CDI.current()` has to look up the CDI container first. Injecting an `Instance` would be better. I'll change that here.

The other place I used `CDI.current()` was in the `AbstractMediator` which isn't a CDI bean, so we'd have to pass the instance in there instead, or change the place that the decoration is done.

I'll have a poke around and see if I can get rid of the calls to `CDI.current()` as I don't really like them.

Reply:
I've pushed a new commit to inject an instance instead of using `CDI.current()`

Reply:
There were a couple of things I wasn't sure about:
1. I've used a setter on AbstractMediator to pass in the decorators, which must be called. At the moment it would NPE when applying the decorators if it wasn't set. We could add a null check first or pass it through the constructor instead (although then you need to pass it in through the `MediatorFactory` instead and pass it down through the constructor of each subclass).
2. Your `AbstractMediator.setInvoker()` method is `synchronized` and I couldn't work out why that was necessary but it made be concerned that it might also be necessary for `setDecorators()`.

Reply:
Not sure why it's synchronizer. I need to check.
I believe the usage of a setter method is good, we can always revisit this later if we found an issue.